ref_arch/multi_tenant.rst

```diff
@@ -0,0 +1,511 @@
+
+Node addition takes around five minutes. Refresh the browser until the change-in-progress message disappears. Next select the "Nodes" tab in the Cloud Console. You should see three nodes listed. Notice how the new node has no data on it (data size = 0 bytes):
+
+.. image:: ../images/cloud-node-stats.png
+
+To bring the node into play we can ask Citus to rebalance the data. This operation moves bundles of rows called shards between the currently active nodes to attempt to equalize the amount of data on each node. Rebalancing preserves :ref:`colocation`, which means we can tell Citus to rebalance the :code:`companies` table and it will take the hint and rebalance the other tables which are distributed by :code:`company_id`.
+
+Applications do not need to undergo downtime during shard rebalancing. Read requests continue seamlessly, and writes are locked only when they affect shards which are currently in flight.
+
+::
+
+  -- Spread data evenly between the nodes
+
+  SELECT rebalance_table_shards('companies');
+
+As it executes, the command outputs notices of each shard it moves:
+
+.. code-block:: text
+
+  NOTICE:  00000: Moving shard [id] from [host:port] to [host:port] ...
+
+Refreshing the Nodes tab in the Cloud Console shows that the new node now contains data! It can now help processing requests for some of the tenants.
+
+Dealing with Big Tenants
+------------------------
+
+.. note::
+
+  This section uses features available in Citus Cloud and Citus Enterprise, and will not work in the community edition.
+
+The previous section describes a general-purpose way to scale a cluster as the number of tenants increases. However, users often have two concerns. The first is that their largest tenant won't be able to fit on a single node. The second is that hosting a large tenant together with small ones on a single worker node can degrade the performance for all of them.
+
+The first concern tends to be unwarranted. Investigating data from large SaaS sites reveals that as the number of tenants increases, the size of tenant data typically tends to follow a `Zipfian distribution <https://en.wikipedia.org/wiki/Zipf%27s_law>`_. For instance in a database of 10 tenants, the largest is predicted to account for 60% of the data. When there are 10k tenants, the largest will account for 2% of all data. Even at 10TB for 10k tenants, the largest tenant will require 200GB, which can still fit on a single node.
+
+There's truth to the second concern, but there's a solution as well. The true part is that standard Citus shard rebalancing will improve overall performance but it may or may not improve the mixing of large and small tenants. The rebalancer simply distributes shards to equalize storage usage on nodes, without examining which tenants are allocated on each shard.
```

Comment:
Slight re-write: "Another question is regarding performance when large and small tenants are on the same node. Standard shard rebalancing ....."